pilot/pkg/bootstrap/server.go

```diff
@@ -1183,7 +1184,64 @@ func (s *Server) initMeshHandlers() {
 	})
 }
 
-func (s *Server) initWorkloadTrustBundle() {
+func (s *Server) addIstioCAToTrustBundle(args *PilotArgs) {
+	// TODO: unify with CreateIstioCA instead of mirroring it
+
+	var err error
+	if s.CA != nil {
+		// If IstioCA is setup, derive trustAnchor directly from CA
+		rootCerts := []string{string(s.CA.GetCAKeyCertBundle().GetRootCertPem())}
+		err = s.workloadTrustBundle.UpdateTrustAnchor(&tb.TrustAnchorUpdate{
+			TrustAnchorConfig: tb.TrustAnchorConfig{Certs: rootCerts},
+			Source:            tb.SourceIstioCA,
+		})
+		if err != nil {
+			log.Errorf("unable to add CA root as trustAnchor")
+		}
+		return
+	}
+
+	// If CA is not running, derive root certificates from the configured CA secrets
+	signingKeyFile := path.Join(LocalCertDir.Get(), "ca-key.pem")
+	if _, err := os.Stat(signingKeyFile); err != nil && s.kubeClient != nil {
+		// Fetch self signed certificates
+		caSecret, scrtErr := s.kubeClient.CoreV1().Secrets(args.Namespace).
+			Get(context.TODO(), ca.CASecret, metav1.GetOptions{})
+		if scrtErr != nil {
+			log.Errorf("unable to retrieve self signed CA secret: %v", scrtErr)
```

Comment:
should be a fatal error or retry loop. we cannot silently ignore this error or we may get inconsistent state